gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/RequestHandler.java

```diff
@@ -88,16 +88,21 @@ public RequestHandler(
   @Override
   public ExploreResponse.Builder handleRequest(
       ExploreRequestContext requestContext, ExploreRequest request) {
-    QueryRequest queryRequest =
+    Optional<QueryRequest> maybeQueryRequest =
         buildQueryRequest(requestContext, request, attributeMetadataProvider);
 
+    if (maybeQueryRequest.isEmpty()) {
+      return ExploreResponse.newBuilder();
+    }
+
+    QueryRequest queryRequest = maybeQueryRequest.get();
```

Comment:
nit - prefer `.orElseThrow` over `.get` (or move code around to avoid the direct access)